compiler/damlc/daml-stdlib-src/DA/NonEmpty.daml

```diff
@@ -40,6 +40,9 @@ import DA.Action qualified as M
 deriving instance Eq a => Eq (NonEmpty a)
 deriving instance Show a => Show (NonEmpty a)
 
+instance (Eq a, Ord a) => Ord (NonEmpty a) where
+  compare x y = compare (toList x) (toList y)
```

Comment:
```suggestion
deriving instance Ord a => Ord (NonEmpty a)
```
We can just derive this instance similar to how we handle Eq and Show above.

Two small comments on your hand-written instance:
1. Eq is a superclass of Ord so the constraint is redundant.
2. `compare (toList x) (toList y)` is somewhat inefficient. There is no fusion or laziness in Daml so this has to build two new lists before comparing them. While that’s not super expensive it’s still unnecessary. The efficient way to write this is something like
```
compare x y = case compare (hd x) (hd y) of
  EQ -> compare (tl x) (tl y)
  LT -> LT
  GT -> GT
```
Luckily that’s exactly what the autoderived instance does so no need to write this ourselves.

Reply:
Thanks for the suggestion. My thinking was to keep the Ord instance behaviour always the same as the behaviour for List, but because of the way NonEmpty is implemented the derived instance will do the same and more efficiently.